service/matching/matcher.go

```diff
@@ -440,3 +538,80 @@ func (tm *TaskMatcher) fwdrAddReqTokenC() <-chan *ForwarderReqToken {
 func (tm *TaskMatcher) isForwardingAllowed() bool {
 	return tm.fwdr != nil
 }
+
+// isBacklogNegligible returns true of the age of backlog is less than the threshold. Note that this relies on incoming
+// MustOffer calls being made back-to-back. If called between two subsequent MustOffer calls, it may return false positive.
```

Comment:
do you think we should add a comment to MustOffer explaining this restriction?

I did have the idea of offering a few tasks from the backlog at once to help with some of these issues. is that an idea that's still possibly helpful, or do we definitely not need it anymore?